content/en/blog/_posts/2023-12-14-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,213 @@
+> component flag.
+
+> `DisableKubeletCloudCredentialProviders`: Disable the in-tree functionality
+> in kubelet to authenticate to a cloud provider container registry for image
+> pull credentials.
+
+## What do you need to do?
+
+If you are upgrading from Kubernetes 1.28+ and are not on Azure, GCE, or
+vSphere then there are no changes you will need to make. If
+you **are** on Azure, GCE, or vSphere, or you are upgrading from a version
+older than 1.28, then read on.
+
+Historically, Kubernetes has included code for a set of cloud providers that
+included AWS, Azure, GCE, OpenStack, and vSphere. Since the inception of
+[KEP-2395][kep2395] the community has been moving towards removal of that
+cloud provider code. The OpenStack provider code was removed in version 1.26,
+and the AWS provider code was removed in version 1.27. This means that users
+who are upgrading from one of the affected cloud providers and versions will
+need to modify their deployments.
+
+### Upgrading on Azure, GCE, or vSphere
+
+There are two options for upgrading in this configuration: migrate to external
+cloud controller managers, or continue using the in-tree provider code.
+Although migrating to external cloud controller managers is recommended,
+there are scenarios where continuing with the current behavior is desired.
+Please choose the best option for your needs.
+
+#### Migrate to external cloud controller managers
+
+Migrating to use external cloud controller managers is the recommended upgrade
+path, when possible in your situation. To do this you will need to
+enable the `--cloud-provider=external` command line flag for the
+`kube-apiserver`, `kube-controller-manager`, and `kubelet` components. In
+addition you will need to deploy a cloud controller manager for your provider.
+
+Installing and running cloud controller managers is a larger topic than this
+post can address; if you would like more information on this process please
+read the documentation for [Cloud Controller Manager Administration][ccmadmin]
+and [Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmha].
+See [below](#cloud-provider-integrations) for links to specific cloud provider
+implementations.
+
+#### Continue using in-tree provider code
+
+If you wish to continue using Kubernetes with the in-tree cloud provider code,
+you will need to modify the command line parameters for `kube-apiserver`,
+`kube-controller-manager`, and `kubelet` to disable the feature gates for
+`DisableCloudProviders` and `DisableKubeletCloudCredentialProviders`. To do
+this, add the following command line flag to the arguments for the previously
+listed commands:
+
+```
+--feature-gates=DisableCloudProviders=false,DisableKubeletCloudCredentialProviders=false
+```
+
+_Please note that if you have other feature gate modifications on the command
+line, they will need to include these 2 feature gates._
+
+**WARNING**: These feature gates will be locked to `true` in an upcoming
```

Comment:
```suggestion
**NOTE**: These feature gates will be locked to `true` in an upcoming
```

“Warning” is reserved for advice that people **must** follow right now or they risk a serious problem (security breach, cluster outage, etc). This is important but not yet that serious.

You could also use “caution” if you'd like to; that's less serious than Warning (in our docs).